lib/ansible/modules/cloud/amazon/ecs_service.py

```diff
@@ -458,10 +475,19 @@ def main():
         module.fail_json(msg="Exception describing service '" + module.params['name'] + "' in cluster '" + module.params['cluster'] + "': " + str(e))
 
     results = dict(changed=False)
+
+    if module.params['launch_type']:
+        if not service_mgr.ecs_api_supports_launch_type():
+            module.fail_json(msg='botocore needs to be version 1.8.4 or higher to use launch_type')
+
     if module.params['state'] == 'present':
 
         matching = False
         update = False
+
+        if module.params['launch_type'] == 'FARGATE' and ('network_configuration' not in module.params or not module.params['network_configuration']):
```

Comment:
I think this can just be added to the require_if on [line 439](https://github.com/ansible/ansible/pull/40736/files#diff-5ab98c93ae8d77e786cf0afa8a0e95c3L439) with `('launch_type', 'FARGATE', ['network_configuration'])`